x-pack/plugins/security_solution/cypress/e2e/investigations/timelines/discover/search_filter.cy.ts

```diff
@@ -0,0 +1,137 @@
+      gotToDiscoverTab();
+      updateDateRangeInLocalDatePickers(DISCOVER_CONTAINER, INITIAL_START_DATE, INITIAL_END_DATE);
+    });
+    it('should change data when dataView is changed', () => {
+      switchDataViewTo('.kibana-event-log');
+      cy.get(DISCOVER_RESULT_HITS).should('have.text', '1');
+    });
+
+    it('should show data according to kql query', () => {
+      const kqlQuery = '_id:"invalid"';
+      addDiscoverKqlQuery(kqlQuery);
+      submitDiscoverSearchBar();
+      cy.get(DISCOVER_NO_RESULTS).should('be.visible');
+    });
+    it('should show correct data according to filter applied', () => {
+      openAddDiscoverFilterPopover();
+      fillAddFilterForm({
+        key: 'agent.type',
+        value: 'winlogbeat',
+      });
+      cy.get(DISCOVER_FILTER_BADGES).should('have.length', 1);
+      cy.get(DISCOVER_RESULT_HITS).should('have.text', '1');
+    });
+    it('should show correct data according to query DSL', () => {
+      const query = {
+        bool: {
+          filter: [
+            {
+              term: {
+                'agent.type': 'winlogbeat',
+              },
+            },
+          ],
+        },
+      };
+      openAddDiscoverFilterPopover();
+      fillAddFilterFormAsQueryDSL(JSON.stringify(query));
+      cy.get(DISCOVER_FILTER_BADGES).should('have.length', 1);
+      cy.get(DISCOVER_RESULT_HITS).should('have.text', '1');
+    });
+
+    it('should be able to create ad-hoc dataview without saving', () => {
+      const adHocDVName = 'adHocDataView';
+      const indexPattern = 'audit';
+      createAdHocDataView(adHocDVName, indexPattern);
+      cy.get(DISCOVER_DATA_VIEW_SWITCHER.BTN).should('contain.text', adHocDVName);
+    });
+
+    it('should be able to add fields to the table', () => {
+      addFieldToTable('host.name');
+      cy.get(GET_DISCOVER_DATA_GRID_CELL_HEADER('host.name')).should('be.visible');
+      addFieldToTable('user.name');
+      cy.get(GET_DISCOVER_DATA_GRID_CELL_HEADER('user.name')).should('be.visible');
+    });
+
+    context('navigation', () => {
+      it('should remove the filter when back is pressed after adding a filter', () => {
+        openAddDiscoverFilterPopover();
+        fillAddFilterForm({
+          key: 'agent.type',
+          value: 'winlogbeat',
+        });
+        cy.get(DISCOVER_FILTER_BADGES).should('have.length', 1);
+        cy.go('back');
+        cy.get(DISCOVER_FILTER_BADGES).should('not.exist');
+      });
+      it('should removed the query when back is pressed after adding a query', () => {
+        const kqlQuery = '_id:"invalid"';
+        addDiscoverKqlQuery(kqlQuery);
+        submitDiscoverSearchBar();
+        cy.get(DISCOVER_QUERY_INPUT).should('have.text', kqlQuery);
+        cy.go('back');
+        cy.get(DISCOVER_QUERY_INPUT).should('not.have.text', kqlQuery);
+      });
+
+      it('should changed the timerange to foo when back is pressed after modifying timerange from foo to baz ', () => {
```

Comment:
nit: mind replacing `foo` and `baz` with the actual dates `${INITIAL_START_DATE}` ${NEW_START_DATE}`?

Reply:
@michaelolo24  Done.